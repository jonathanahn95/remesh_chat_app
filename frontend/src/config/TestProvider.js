import configureMockStore from 'redux-mock-store';
import thunkMiddleWare from 'redux-thunk';

export const mockStore = configureMockStore([thunkMiddleWare])
