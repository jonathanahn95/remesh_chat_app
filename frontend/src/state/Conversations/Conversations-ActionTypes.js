export const SET_CONVERSATIONS_REQUESTS = 'SET_CONVERSATIONS_REQUESTS';
export const SET_CREATED_CONVERSATION_REQUEST = 'SET_CREATED_CONVERSATION_REQUEST';
export const FETCH_CONVERSATION = 'FETCH_CONVERSATION';
