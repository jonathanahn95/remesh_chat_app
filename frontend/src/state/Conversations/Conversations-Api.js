export async function fetchConversation(id) {
  return await fetch(`/api/v1/conversations/${id}`)
  .then(response => response.json())
  .catch((error) => {
    console.error('Error:', error)
  })
};

export async function getConversationsRequest() {
  return await fetch('/api/v1/conversations')
  .then(response => response.json())
  .catch((error) => {
    console.error('Error:', error)
  })
};

export async function createConversation(data) {
  const rawData = await fetch('/api/v1/conversations', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(data)
  })
 
  const content = await rawData.json()
  return content;
};
