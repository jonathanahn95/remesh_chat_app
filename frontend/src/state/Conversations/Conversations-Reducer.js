import { SET_CONVERSATIONS_REQUESTS, SET_CREATED_CONVERSATION_REQUEST, FETCH_CONVERSATION } from './Conversations-ActionTypes';

const INITIAL_STATE = {
    conversations: [],
    conversation: {},
    error: null,
};

export default function conversationsReducer(state = INITIAL_STATE , action) { 
    switch (action.type) { 
        case FETCH_CONVERSATION: 
            return {
                ...state,
                conversation: action.payload.data,
                error: ""
            }
        case SET_CONVERSATIONS_REQUESTS: 
            return {
                ...state,
                conversations: action.payload.data,
                error: ""
            }
        case SET_CREATED_CONVERSATION_REQUEST: 
            return {
                ...state,
                conversations: [...state.conversations, action.payload.data],
                error: ""
            }
        default: 
            return state;
    }
}

