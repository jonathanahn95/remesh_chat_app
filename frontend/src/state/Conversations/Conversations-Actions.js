import { GET_ALL_CONVERSATIONS_SUCCESS, CREATE_CONVERSATION_SUCCESS, GET_SINGLE_CONVERSATION_SUCCESS, CREATE_CONVERSATION_FAILURE } from './Conversations-ActionTypes'
import * as ConversationApi from "./Conversations-Api";



export const setSingleConversation = payload => {
    return {
      type: GET_SINGLE_CONVERSATION_SUCCESS,
      payload
    };
};

const setAllConversations = payload => {
    return {
      type: GET_ALL_CONVERSATIONS_SUCCESS,
      payload
    };
};

const setCreatedConversation = payload => {
    return {
      type: CREATE_CONVERSATION_SUCCESS,
      payload
    };
};

const setCreatedConversationFailure = payload => {
    return {
      type: CREATE_CONVERSATION_FAILURE,
      payload
    };
};

export const getSingleConversation = (id) => {
  return dispatch => {
    return ConversationApi.getSingleConversation(id).then(conversation => {
      return dispatch(setSingleConversation(conversation));
    });
  };
};

export const getAllConversations = () => {
  return dispatch => {
    return ConversationApi.getAllConversations().then(conversations => {
      return dispatch(setAllConversations(conversations));
    });
  };
};


export const createConversation = (data) => async (dispatch) => {
  let obj = data
  try {
    const response = await fetch('/api/v1/conversations', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(obj)
    });
    const res = await response
    const jsonRes = await response.json()
    if (res.status === 200) {

      dispatch(setCreatedConversation(jsonRes))
    } else {
      dispatch(setCreatedConversationFailure(res.error))
    }
  } catch (e) {
    dispatch(setCreatedConversationFailure(e.message))
  }
};