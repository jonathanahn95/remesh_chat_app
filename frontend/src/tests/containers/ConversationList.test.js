import { shallow, mount } from 'enzyme';
import { render } from '@testing-library/react';
import React from 'react';
import ConversationList from '../../containers/ConversationsList';
import TestProvider from '../../utils/index';
import configureMockStore from 'redux-mock-store';
import thunkMiddleWare from 'redux-thunk';
import renderer from 'react-test-renderer';
import { createConversation } from '../../state/Conversations/Conversations-Actions'


const mockStore = configureMockStore([thunkMiddleWare])

const conversation = {
    id: '1',
    type: 'conversation',
    attributes: {
        title: 'General'
    },
    relationships: {
        messages: [
            {id: 1, type: 'message'}
        ]
    }
}


describe('The ConversationList Component', () => {
    let store;
    let component;

    beforeEach(() => {
      store = mockStore({
        myState: 'sample text',
      });

      store.dispatch = jest.fn();
       
    component = renderer.create(
        <TestProvider store={store}>
          <ConversationList />
        </TestProvider>
      );
    });

    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
})