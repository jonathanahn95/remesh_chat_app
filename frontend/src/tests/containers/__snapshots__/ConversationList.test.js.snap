// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ConversationList Component should render with given state from Redux store 1`] = `
<div
  className="ConversationsList-root-1"
>
  <h1>
    Channels:
  </h1>
  <form
    className="ConversationsList-form-4"
    onSubmit={[Function]}
  >
    <div
      className="ConversationsList-formSection-5"
    >
      <div
        className="ConversationsList-formTitle-7"
      >
        Add A Conversation:
      </div>
      <div
        className="ConversationsList-buttonSection-6"
      >
        <input
          onChange={[Function]}
          placeholder="Enter a title"
          value=""
        />
        <button
          className="ConversationsList-button-8"
          role="submit"
        >
          Create Conversation
        </button>
      </div>
    </div>
  </form>
  <div>
    No Conversations Found.
  </div>
</div>
`;
